### PR TITLE
ECMA 6 - Destructuring assignment tests

### PR DESCRIPTION
**Pull request by pemattio**
From: https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Operadores/Destructuring_assignment

Work for the following issue:

#126

### CODE REVIEW

File: test/suites/ecma6-tests/ecma6-destructuring-assignment-tests.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        assert.equal(result.value, true);
+	});
+	
+	skip("Object destructuring: function arguments default, simple example", function(assert) {
+        var result = puma.evalPuma("let {x = 1, y = 200} = {x: 100};");
+        assert.equal(result.value, true);
+	});	
+	
+	skip("Object destructuring: function arguments default", function(assert) {
+        var result = puma.evalPuma("function drawES2015Chart({size = 'big', cords = { x: 0, y: 0 }, radius = 25} = {}){return [size, cords, radius]} let [size, cords, radius] = drawES2015Chart({cords: { x: 18, y: 30 }, radius: 30}); size === 'big' && cords.x === 18 && cords.y === 30 && radius === 30;");
```
Comment (emravera) [inline test/suites/ecma6-tests/ecma6-destructuring-assignment-tests.js]: Please review the complexity of the test and try to reduce to the minimal expression.

File: test/suites/ecma6-tests/ecma6-destructuring-assignment-tests.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        assert.equal(result.value, true);
+	});
+	
+	skip("Object destructuring: nested objects and array destructuring", function(assert) {
+        var result = puma.evalPuma("let metadata = {title: 'Scratchpad', translations: [{locale: 'de', localization_tags: [ ], last_edit: '2014-04-14T08:43:37', url: '/de/docs/Tools/Scratchpad', title: 'JavaScript-Umgebung'}], url: '/en-US/docs/Tools/Scratchpad'}; let { title: englishTitle, translations: [{ title: localeTitle }] } = metadata; englishTitle === 'Scratchpad' && localeTitle === 'JavaScript-Umgebung';");
```
Comment (emravera) [inline test/suites/ecma6-tests/ecma6-destructuring-assignment-tests.js]: Reduce this too.

File: test/suites/ecma6-tests/ecma6-destructuring-assignment-tests.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        assert.equal(result.value, true);
+	});
+	
+	skip("Object destructuring: For iterations with destructuring", function(assert) {
+        var result = puma.evalPuma("var people = [{name: 'Mike Smith',family: {mother: 'Jane Smith', father: 'Harry Smith'}, age: 35}, {name: 'Tom Jones', family: {mother: 'Norah Jones', father: 'Richard Jones'}, age: 25}]; let fathers=[]; for (let {name: n, family: { father: f } } of people) {father=[{name: n, father: f}];fathers.push(father);} fathers.length === 2;");
```
Comment (emravera) [inline test/suites/ecma6-tests/ecma6-destructuring-assignment-tests.js]: This test also can be simpler.

Comment (Undre4m) [reply test/suites/ecma6-tests/ecma6-destructuring-assignment-tests.js]: I agree with the test being a bit difficult to read. IMO indenting the code would solve that problem. We can´t use template literals for multiline comments right?

File: test/suites/ecma6-tests/ecma6-destructuring-assignment-tests.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+	skip("Object destructuring: Computed object property names and destructuring", function(assert) {
+        var result = puma.evalPuma("let key = 'z'; let { [key]: foo } = { z: 'bar' }; foo === 'bar';");
+        assert.equal(result.value, true);
+	});
+	
+	skip("Object destructuring: Computed object property names and destructuring, alternative", function(assert) {
+        var result = puma.evalPuma("let {['f' + 'oo']: there} = {foo : 'bar'}; there === 'bar';");
+        assert.equal(result.value, true);
+	});	
+
```
Comment (Undre4m) [inline test/suites/ecma6-tests/ecma6-destructuring-assignment-tests.js]: Indentation doesn't look right. ESLint isn't running on this folder, i'll be fixing that soon

File: test/suites/ecma6-tests/ecma6-destructuring-assignment-tests.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    });
+
+    skip("Object destructuring: function arguments default", function(assert) {
+        var result = puma.evalPuma("function drawES2015Chart({cords = { x: 0, y: 0 }, radius = 25} = {}){return [cords, radius]} let [cords, radius] = drawES2015Chart({cords: { x: 18, y: 30}}); cords.x;");
+        result.makeValue();
+        assert.ok(result.success && result.value === 18);
+    });
+
+    skip("Object destructuring: nested objects and array destructuring", function(assert) {
+        var result = puma.evalPuma("let metadata = { title: 'Scratchpad', translations: [{ locale: 'de', title: 'JavaScript-Umgebung'}] }; let { title: englishTitle, translations: [{ title: localeTitle }] } = metadata; localeTitle;");
```
Comment (emravera) [inline test/suites/ecma6-tests/ecma6-destructuring-assignment-tests.js]: Review let for const

File: test/suites/ecma6-tests/ecma6-destructuring-assignment-tests.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        result.makeValue();
+        assert.ok(result.success && result.value === 'JavaScript-Umgebung');
+    });
+
+    skip("Object destructuring: For iterations with destructuring", function(assert) {
+        var result = puma.evalPuma("let people = [{ name: 'Mike Smith', family: { father: 'Harry Smith' } }, { name: 'Tom Jones', family: { father: 'Richard Jones' } }]; let fathers = []; for (let { family: { father: f } } of people) {father = [{ father: f }]; fathers.push(father);} fathers.length;");
```
Comment (emravera) [inline test/suites/ecma6-tests/ecma6-destructuring-assignment-tests.js]: Review let and const

File: test/suites/ecma6-tests/ecma6-destructuring-assignment-tests.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        assert.ok(result.success && result.value === 'bar');
+    });
+
+    skip("Object destructuring: Computed object property names and destructuring, alternative", function(assert) {
+        var result = puma.evalPuma("let {['f' + 'oo']: there} = {foo : 'bar'}; there;");
+        result.makeValue();
+        assert.ok(result.success && result.value === 'bar');
+    });
+
+});
```
Comment (emravera) [inline test/suites/ecma6-tests/ecma6-destructuring-assignment-tests.js]: Add blank line.

**Pull request merged**